common/crypto/verify_ias_report/fetch_ias_certificates.sh

```diff
@@ -0,0 +1,69 @@
+#!/bin/bash
+# Copyright 2023 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This script sets up the IAS root certificate for inclusing in the
+# verification module.
+#
+# Two parameters:
+# $1 -- the URL where the IAS certificate can be retrieved
+# $2 -- the file where the certificate should be written
+
+# -----------------------------------------------------------------
+# -----------------------------------------------------------------
+source ${PDO_SOURCE_ROOT}/bin/lib/common.sh
+
+IAS_CERTIFICATE_URL=$1
+
+# -----------------------------------------------------------------
+# set up the temporary files
+# -----------------------------------------------------------------
+SAVE_FILE=$(mktemp /tmp/pdo-ias-certificate.XXXXXXXXX)
+STRING_FILE=$(mktemp /tmp/pdo-ias-certificate-string.XXXXXXXXX)
+
+function cleanup {
+    rm -f ${SAVE_FILE} ${STRING_FILE}
+}
+
+trap 'echo "**ERROR - line $LINENO**"; cleanup; exit 1' HUP INT QUIT PIPE TERM ERR
+
+# If there is no requirement for HW support, then we don't need
+# a valid certificate; just generate a dummy string
+if [ "${SGX_MODE}" != "HW" ]; then
+    echo 'R"IASCERT(' > ${STRING_FILE}
+    echo 'NO CERTIFICATE REQUIRED' >> ${STRING_FILE}
+    echo ')IASCERT"' >> ${STRING_FILE}
+
+    try mv ${STRING_FILE} $2
+    exit
+fi
+
+# -----------------------------------------------------------------
+# get the certificate and format it as needed
+# -----------------------------------------------------------------
+
+# This is a small hack to make the script work for people
+# who would otherwise attempt to retrieve the certficiates
+# without a proxy server
```

Comment:
Text seems a bit misleading?  This is really only for folks (a) inside intel  _with_ a proxy server but (b) define `no_proxy` generically as `intel.com` rather more narrowly as, e.g., `.jf.intel.com,10.0.0.0/8,134.134.0.0/16,localhost,.local,127.0.0.0/8,172.16.0.0/12,192.168.0.0/16,.devtools.intel.com,.iglb.intel.com,isscorp.intel.com,.corp.intel.com,.caas.intel.com,certificates.intel.com` ?

Reply:
its not possible with no-proxy alone to enumerate all possible hosts that *might* be internal. you can certainly add in some portion of the correct hosts & make it work. or you can just set this to force access through the proxy.

Reply:
I guess with my ramble in the second part i hid the main point i wanted to say is that the hack is specific to intel folks in some circumstances.  Hopefully, below text makes it a bit clearer.
```suggestion
# inside  intel who would otherwise attempt to retrieve the certficiates
# without a proxy server due to `no_proxy=intel.com` or alike
```

Reply:
> I guess with my ramble in the second part i hid the main point i wanted to say is that the hack is specific to intel folks in some circumstances. Hopefully, below text makes it a bit clearer.

I specifically do not want to reference intel in any external code. We KNOW that its there for us. But the code is intended to be a general (if not very important under most circumstances) solution.